packages/react-reconciler/src/ReactFiberCompleteWork.new.js

```diff
@@ -696,6 +699,19 @@ function bubbleProperties(completedWork: Fiber) {
   let subtreeFlags = NoFlags;
 
   if (!didBailout) {
+    if (enableDoubleInvokingEffects) {
+      if (__DEV__) {
+        if (completedWork.alternate === null) {
+          if ((completedWork.flags & PassiveMask) !== NoFlags) {
+            subtreeFlags |= MountPassiveDev;
+          }
+          if ((completedWork.flags & LayoutMask) !== NoFlags) {
+            subtreeFlags |= MountLayoutDev;
+          }
+        }
```

Comment:
Seems like this special casing shouldn't be required. We should just assign the new flags during mount, when we assign `Passive` or `Layout`. I think doing this would be a little more work (more lines of code) but it would let us remove the redundant `current === null` check in `invokeEffectsInDev` and also let us pass in a single `fiberFlags` param rather than two.